flink-runtime/src/test/java/org/apache/flink/runtime/taskmanager/TaskTest.java

```diff
@@ -922,6 +926,40 @@ public void testTerminationFutureCompletesOnErrorInInvoke() throws Exception {
 		assertEquals(ExecutionState.FAILED, task.getTerminationFuture().getNow(null));
 	}
 
+	@Test
```

Comment:
Either instead of those unit tests or additionally to them, could we add an ITCase which would validate that some simple `DataStream` application with a source and not chained task correctly reports a backpressure?

Reply:
Done. I think we can delete `BackPressureStatsTrackerImplITCase`. WDYT?

Reply:
I'm not entirely sure. Technically speaking with new test we are only checking for the back pressure presence in `DataStream` api. However old test `BackPressureStatsTrackerImplITCase` was not doing much, since it was asserting that custom task manually invokes `requestBufferBuilderBlocking`, which I would say is superseded and covered even better by the new test (since it's at least using production `StreamTask` class).

So I think I would drop the old test, but it's a pitty that we do not have a coverage for `DataSet` api.

Reply:
Let's delete it.

Reply:
> Either instead of those unit tests or additionally to them [...]

I decided to drop the unit tests here except for `testReturnsEmptyStackTraceIfTaskNotStarted` because
- there was a race condition in `testReturnsStackTraceOfTaskExecutingThread` and `testReturnsStackTraceOfAbstractInvokableExecutingThread`
- they are covered by `BackPressureITCase`

Reply:
`BackPressureStatsTrackerImplITCase` is deleted